src/zope/security/checker.py

```diff
@@ -770,6 +770,43 @@ def f(): #pragma NO COVER
     _default_checkers[type({}.iterkeys())] = _iteratorChecker
     _default_checkers[type({}.itervalues())] = _iteratorChecker
 
+try:
+    import BTrees
+except ImportError: # pragma: no cover
+    pass
+else:
+    # The C implementation of BTree.items() is its own iterator
+    # and doesn't need any special entries to enable iteration.
+    # But the Python implementation has to call __iter__ to be able
+    # to do iteration. Whitelist it so that they behave the same.
+    # In addition, Python 3 will attempt to call __len__ on iterators
+    # for a length hint, so the C implementations also need to be
+    # added to the _iteratorChecker.
+    # We do this here so that all users of zope.security can benefit
+    # without knowing implementation details.
+    # See https://github.com/zopefoundation/zope.security/issues/20
```

Comment:
That is excellent news (and maybe worth mentioning in the comment/added to the test)!